website/blog/2025-06-05-replacing-mongodb-with-ferretdb-librechat.md

```diff
@@ -0,0 +1,263 @@
+
+For LibreChat users who want to stay fully open source, FerretDB is a great drop-in replacement for MongoDB,
+especially if you're looking to avoid proprietary databases or vendor lock-in.
+It uses PostgreSQL with DocumentDB extension as the backend, while letting you use familiar MongoDB operations and commands.
+
+This guide shows how to run LibreChat with FerretDB (instead of MongoDB) and Ollama to run the `phi4-mini` model locally – with all components being fully open source.
+
+## Prerequisites
+
+To follow this guide, ensure:
+
+- [Docker](https://www.docker.com/) is installed and running on your machine.
+- Ollama is installed and running on your machine.
+  If you haven't installed Ollama yet, download it from the [official website](https://ollama.com/download).
+  This guide uses the `phi4-mini` model, an open-source model optimized for reasoning and accuracy in text-based tasks.
+  Once Ollama is running, pull the `phi4-mini` model:
+
+  ```sh
+  ollama pull phi4-mini
+  ```
+
+## How to set up LibreChat with FerretDB
+
+Start by cloning the LibreChat repository:
+
+```sh
+git clone https://github.com/danny-avila/LibreChat.git
+cd LibreChat
+```
+
+You can find more instructions on how to set up LibreChat in the [LibreChat documentation](https://www.librechat.ai/docs/quick_start/local_setup).
+
+Next, copy the `.env.example` file to `.env`:
+
+```sh
+cp .env.example .env
+```
+
+You may need to adjust the environment variables in `.env` to suit your setup.
+For instance, you may have set the `UID` and `GID` to `999` in the `.env` file, as it was the case in this guide.
+
+You can run FerretDB alongside LibreChat using Docker Compose.
+
+To do that, we'll add the [FerretDB evaluation image](https://docs.ferretdb.io/installation/evaluation/) to your `docker-compose.override.yml` file, as shown below:
+
+```yaml
+services:
+  api:
+    environment:
+      - MONGO_URI=mongodb://<username>:<password>@ferretdb:27017/LibreChat
+    depends_on:
+      - ferretdb
+
+  ferretdb:
+    image: ghcr.io/ferretdb/ferretdb-eval:2
+    restart: on-failure
+    ports:
+      - 27017:27017
+    environment:
+      - POSTGRES_USER=<username>
+      - POSTGRES_PASSWORD=<password>
+      - POSTGRES_DB=postgres
+
+  mongodb:
+    profiles:
+      - donotstart
+```
+
+Replace `<username>` and `<password>` with your desired FerretDB credentials.
+
+In the above `docker-compose.override.yml` file, we use the FerretDB evaluation image (`ferretdb-eval:2`), which is suitable for development and testing purposes.
+It includes PostgreSQL with DocumentDB extension, which FerretDB uses as its backend.
+
+We also place the `mongodb` service under the `donotstart` profile so it won't start by default.
+This means you need to explicitly update the LibreChat `api` service to not depend on the `mongodb` service in your `docker-compose.yml` file – or you can remove the `mongodb` service entirely from the `docker-compose.yml` file.
```

Comment:
I don't think explicitly updating is necessary if we are using override file right? Can we replace `depends_on` attribute? https://docs.docker.com/reference/compose-file/merge/#replace-value. Does that work?

```
services:
  api:
    depends_on: !override
      - ferretdb
```